test/integration/taskRequests.test.js

```diff
@@ -0,0 +1,286 @@
+const chai = require("chai");
+const sinon = require("sinon");
+const { expect } = chai;
+const chaiHttp = require("chai-http");
+
+const app = require("../../server");
+const authService = require("../../services/authService");
+const tasksModel = require("../../models/tasks");
+const taskRequestsModel = require("../../models/taskRequests");
+const addUser = require("../utils/addUser");
+const cleanDb = require("../utils/cleanDb");
+const userData = require("../fixtures/user/user")();
+const taskData = require("../fixtures/tasks/tasks")();
+chai.use(chaiHttp);
+
+const config = require("config");
+const { TASK_REQUEST_STATUS } = require("../../constants/taskRequests");
+const cookieName = config.get("userToken.cookieName");
+
+let jwt;
+let taskId;
+
+const member = userData[9];
+const member2 = userData[10];
+const superUser = userData[4];
+const activeMember = userData[0];
+
+describe("Task Requests", function () {
+  let userId, userId2;
+  after(async function () {
+    await cleanDb();
+  });
+
+  afterEach(async function () {
+    sinon.restore();
+  });
+
+  describe("GET / - gets tasks requests", function () {
+    describe("When the user is super user", function () {
+      before(async function () {
+        userId = await addUser(member);
+        const superUserId = await addUser(superUser);
+        jwt = authService.generateAuthToken({ userId: superUserId });
+
+        taskId = (await tasksModel.updateTask(taskData[4])).taskId;
+        await taskRequestsModel.createTaskRequest(taskId, userId);
+      });
+
+      it("should fetch taskRequests", function (done) {
+        chai
+          .request(app)
+          .get("/taskRequests")
+          .set("cookie", `${cookieName}=${jwt}`)
+          .end((err, res) => {
+            if (err) {
+              return done(err);
+            }
+
+            expect(res).to.have.status(200);
+            expect(res.body.message).to.equal("Task requests returned successfully");
+            expect(res.body.taskRequests).to.be.a("Array");
+            expect(res.body.taskRequests.length).to.equal(1);
+            return done();
+          });
+      });
+    });
+
+    describe("When the user is not a super user", function () {
+      before(async function () {
```

Comment:
Same, we can do mocking here as well instead of employing/calling acutal function code.